### PR TITLE
feat(recordings): filter by events in session recording

### PR DESCRIPTION
**Pull request by alexkim205**
## Problem

Global playlist filters should identify matching events in session recording lists.

## Changes

From Insight

https://user-images.githubusercontent.com/13460330/192576836-7d4c00ae-6773-4975-aa02-3e5cee5af279.mov

From Recording Playlist

https://user-images.githubusercontent.com/13460330/192576854-96e3fda3-9e41-43ec-98a1-67fd1bb112b2.mov

TODO in a separate PR: Fuzzy matching

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

Logic tests for new filtering logic.

### CODE REVIEW

File: posthog/queries/session_recordings/session_recording_list.py [hunk 1]
```diff
@@ -355,7 +355,7 @@ def _data_to_return(self, results: List[Any]) -> List[Dict[str, Any]]:
                 "matching_events": [
                     {
                         "events": [
-                            dict(zip(["timestamp", "id", "session_id", "window_id"], event)) for event in row[i + 1]
```
Comment (alexkim205) [inline posthog/queries/session_recordings/session_recording_list.py:358]: renaming to uuid to remain consistent with matching events data returned from persons query

File: frontend/src/scenes/session-recordings/player/list/sharedListLogic.ts [hunk 2]
```diff
@@ -30,6 +31,12 @@ export const sharedListLogic = kea<sharedListLogicType>([
                 setTab: (_, { tab }) => tab,
             },
         ],
+        showOnlyMatching: [
```
Comment (rcmarron) [inline frontend/src/scenes/session-recordings/player/list/sharedListLogic.ts:34]: I think we should move this to the `playerSettingsLogic` or somewhere that isn't keyed on the recording. That way, you can toggle this setting and then quickly click through recordings in the playlist, and the setting is persisted

File: frontend/src/types.ts [hunk 3]
```diff
@@ -2050,6 +2055,7 @@ export interface SessionRecordingPlayerProps {
     playerKey: string
     includeMeta?: boolean
     recordingStartTime?: string
+    matching?: MatchedRecording[]
```
Comment (rcmarron) [inline frontend/src/types.ts:2058]: I think this name is a bit vague. Should it be `matchingEvents` and then the type be `MatchedRecordingEvents[]`?

File: frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.test.ts [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+import { eventUsageLogic } from 'lib/utils/eventUsageLogic'
+import { sessionRecordingDataLogic } from 'scenes/session-recordings/player/sessionRecordingDataLogic'
+import { sharedListLogic } from 'scenes/session-recordings/player/list/sharedListLogic'
+import { playerSettingsLogic } from 'scenes/session-recordings/player/playerSettingsLogic'
+import { useMocks } from '~/mocks/jest'
+import recordingSnapshotsJson from 'scenes/session-recordings/__mocks__/recording_snapshots.json'
+import recordingMetaJson from 'scenes/session-recordings/__mocks__/recording_meta.json'
+import recordingEventsJson from 'scenes/session-recordings/__mocks__/recording_events.json'
+
+describe('sessionRecordingPlayerLogic', () => {
```
Comment (rcmarron) [inline frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.test.ts:13]: ❤️

**Review by rcmarron — COMMENTED** [review]
Looking great! A few nits in the comments. Capturing the couple things we just talked about too here, so we don't miss them:

* Info bubble on filter control
* Make the matching event pins colored and move to forefront
* List filter should have an empty default event
* Maybe some weird bug with matching events being empty

File: frontend/src/scenes/session-recordings/player/list/sharedListLogic.ts (hunk 2 above)
Comment (alexkim205) [reply frontend/src/scenes/session-recordings/player/list/sharedListLogic.ts:34]: good idea!

File: frontend/src/types.ts (hunk 3 above)
Comment (rcmarron) [reply frontend/src/types.ts:2058]: I wonder if it should be a dict then? Like `matching.events = []`. If we do want to expand the prop to other types, then we probably don't want those other types interlaced in the same lis.t

**Comment by alexkim205** [comment]
Addressed all four points of these points! + added fuzzy matching 

> * Info bubble on filter control
> * Make the matching event pins colored and move to forefront
> * List filter should have an empty default event
> * Maybe some weird bug with matching events being empty


https://user-images.githubusercontent.com/13460330/192844785-9a8023ba-ce50-4f92-b973-4adf38936d15.mov

**Review by rcmarron — APPROVED** [review]

**Pull request merged**